src/components/ButtonGroupPicker/index.js

```diff
@@ -0,0 +1,150 @@
+class ButtonGroupPicker extends Component {
+    constructor(props) {
+        super(props);
+        this.groupNameId = this.props.name || uniqueId('options');
+        this.errorMessageId = uniqueId('error-message');
+        this.handleOnChange = this.handleOnChange.bind(this);
+    }
+
+    getErrorMessageId() {
+        const { error } = this.props;
+        if (error) {
+            return this.errorMessageId;
+        }
+        return undefined;
+    }
+
+    handleOnChange(event) {
+        const { value: eventValue, checked } = event.target;
+        const { value, multiple, onChange } = this.props;
+
+        if (!multiple) {
+            return onChange(eventValue);
+        }
+
+        if (checked && Array.isArray(value)) {
+            return onChange(value.concat([eventValue]));
+        }
+        if (checked && !Array.isArray(value)) {
+            return onChange([eventValue]);
+        }
+        return onChange(value.filter(valueId => valueId !== eventValue));
+    }
+
+    render() {
+        const {
+            className,
+            style,
+            label,
+            multiple,
+            children,
+            size,
+            error,
+            bottomHelpText,
+            value,
+            required,
+        } = this.props;
+
+        const errorMessageId = this.getErrorMessageId();
+        const context = {
+            onChange: this.handleOnChange,
+            values: value,
+            type: multiple ? 'checkbox' : 'radio',
+            name: this.groupNameId,
+            ariaDescribedBy: errorMessageId,
+            size,
+        };
+
+        return (
+            <StyledContainer className={className} style={style}>
+                <RenderIf isTrue={!!label}>
+                    <StyledLegend>
+                        <RequiredAsterisk required={required} />
+                        {label}
+                    </StyledLegend>
+                </RenderIf>
+                <StyledButtonGroup>
+                    <Provider value={context}>{children}</Provider>
+                </StyledButtonGroup>
+                <RenderIf isTrue={!!bottomHelpText}>
+                    <StyledHelpText>{bottomHelpText}</StyledHelpText>
+                </RenderIf>
+                <RenderIf isTrue={!!error}>
+                    <StyledErrorText id={errorMessageId}>{error}</StyledErrorText>
+                </RenderIf>
+            </StyledContainer>
+        );
+    }
+}
+
+ButtonGroupPicker.propTypes = {
+    /** A CSS class for the outer element, in addition to the component's base classes. */
+    className: PropTypes.string,
+    /** An object with a custom style applied to the outer element. */
+    style: PropTypes.object,
+    /** The title at the top of the component. */
+    label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    /** The name of the ButtonOption selected or if multiple an array of names. */
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.array]).isRequired,
```

Comment:
Normally all props are not required, only carefully discussed props are required.
```suggestion
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
```